### PR TITLE
Update images to Go 1.20

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-5320

### What this PR does / why we need it:
Updates images to Go 1.20, fixes build problems and CI/lint issues from Go 1.20.

### Test plan for issue:

Should be covered by e2e, unit tests, etc

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

**Comment by bennerv** [comment]
@hawkowl can you make sure our images in the Azure image registries we use for ev2 are updated as well?  See the CDPx Container Images wiki page in ADO.

**Comment by hawkowl** [comment]
@bennerv done

**Comment by hawkowl** [comment]
Moved the ancillary changes into https://github.com/Azure/ARO-RP/pull/3413

**Comment by jaitaiwan** [comment]
Will wait to review once 3413 is merged.

**Comment by hawkowl** [comment]
@jaitaiwan should be good to review now

File: cmd/aro/main.go [hunk 1]
```diff
@@ -35,8 +33,6 @@ func usage() {
 }
 
 func main() {
-	rand.Seed(time.Now().UnixNano())
```
Comment (dofinn) [inline cmd/aro/main.go:38]: @hawkowl this should be replaced with `rand.New(rand.NewSource(seed))` as per https://tip.golang.org/doc/go1.20#:~:text=the%20standard%20library.-,math/rand,-The%20math/rand ?

File: .pipelines/onebranch/pipeline.buildrp.official.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ pr: none
 variables:
   Cdp_Definition_Build_Count: $[counter('', 0)] # needed for onebranch.pipeline.version task https://aka.ms/obpipelines/versioning
   ONEBRANCH_AME_ACR_LOGIN: cdpxb8e9ef87cd634085ab141c637806568c00.azurecr.io
-  LinuxContainerImage: $(ONEBRANCH_AME_ACR_LOGIN)/b8e9ef87-cd63-4085-ab14-1c637806568c/official/ubi8/go-toolset:1.18.4 # Docker image which is used to build the project https://aka.ms/obpipelines/containers
+  LinuxContainerImage: $(ONEBRANCH_AME_ACR_LOGIN)/b8e9ef87-cd63-4085-ab14-1c637806568c/official/ubi8/go-toolset:1.20.10 # Docker image which is used to build the project https://aka.ms/obpipelines/containers
```
Comment (dofinn) [inline .pipelines/onebranch/pipeline.buildrp.official.yml:17]: how to verify `.10` ?

File: cmd/aro/main.go (hunk 1 above)
Comment (hawkowl) [reply cmd/aro/main.go:38]: Why? rand.New returns a new rand source, it doesn't modify the global one. As per the go 1.20 docs, it's automatically seeded.

File: .pipelines/onebranch/pipeline.buildrp.official.yml (hunk 2 above)
Comment (dofinn) [reply .pipelines/onebranch/pipeline.buildrp.official.yml:17]: confirmed

**Pull request closed without merging**